config/config_test.go

```diff
@@ -302,3 +384,38 @@ func TestConfigWithSampler(t *testing.T) {
 	require.Equal(t, traceID, sampler.lastTraceID)
 	require.Equal(t, "test", sampler.lastOperation)
 }
+
+func TestNewTracer(t *testing.T) {
+	cfg := &Configuration{ServiceName: "my-service"}
+	_, closer, err := cfg.NewTracer(Metrics(metrics.NullFactory), Logger(log.NullLogger))
+	defer closer.Close()
+
+	require.NoError(t, err)
+}
+
+func TestNewTracerWithoutServiceName(t *testing.T) {
+	cfg := &Configuration{}
+	_, _, err := cfg.NewTracer(Metrics(metrics.NullFactory), Logger(log.NullLogger))
+	require.Error(t, err)
+}
+
+func TestParseTags(t *testing.T) {
+	os.Setenv("existing", "not-default")
+	tags := "key=value,k1=${nonExisting:default}, k2=${withSpace:default},k3=${existing:default}"
+	ts := parseTags(tags)
+	require.Equal(t, 4, len(ts))
+
+	require.Equal(t, "key", ts[0].Key)
```

Comment:
nit: `require` means you cannot continue the test further, should be rarely used, in most cases `assert` is better.